internal/cli/kraft/cloud/instance/create/create.go

```diff
@@ -25,110 +27,119 @@ import (
+		}
 	}
-	if opts.metro == "" {
-		return fmt.Errorf("kraftcloud metro is unset")
+	if opts.Client == nil {
+		opts.Client = kraftcloud.NewInstancesClient(
+			kraftcloud.WithToken(opts.Auth.Token),
+		)
 	}
-	log.G(cmd.Context()).WithField("metro", opts.metro).Debug("using")
-	return nil
-}
 
-func (opts *CreateOptions) Run(ctx context.Context, args []string) error {
-	image := args[0]
-	auth, err := config.GetKraftCloudLoginFromContext(ctx)
-	if err != nil {
-		return fmt.Errorf("could not retrieve credentials: %w", err)
-	}
+	services := []kraftcloudservices.Service{}
 
-	client := kcinstance.NewInstancesClient(
-		kraftcloud.WithToken(auth.Token),
-	)
+	if len(opts.Ports) == 1 && strings.HasPrefix(opts.Ports[0], "443:") && strings.Count(opts.Ports[0], "/") == 0 {
+		split := strings.SplitN(opts.Ports[0], ":", 2)
+		destPort, err := strconv.Atoi(split[1])
+		if err != nil {
+			return nil, fmt.Errorf("invalid external port: %w", err)
+		}
 
-	services := []kcinstance.CreateInstanceServicesRequest{}
+		services = []kraftcloudservices.Service{
+			{
+				Port:            443,
+				DestinationPort: destPort,
+				Handlers: []kraftcloudservices.Handler{
+					kraftcloudservices.HandlerHTTP,
+					kraftcloudservices.HandlerTLS,
+				},
+			},
+			{
+				Port:            80,
+				DestinationPort: 443,
+				Handlers: []kraftcloudservices.Handler{
+					kraftcloudservices.HandlerHTTP,
+					kraftcloudservices.HandlerRedirect,
+				},
+			},
+		}
 
-	if len(opts.Port) > 0 {
-		var internalPort int
-		var externalPort int
-		if strings.ContainsRune(opts.Port, ':') {
-			ports := strings.Split(opts.Port, ":")
-			if len(ports) != 2 {
-				return fmt.Errorf("invalid --port value expected --port <internal>:<external>")
+	} else {
+		for _, port := range opts.Ports {
+			service := kraftcloudservices.Service{
+				Handlers: []kraftcloudservices.Handler{},
 			}
 
-			internalPort, err = strconv.Atoi(ports[0])
-			if err != nil {
-				return fmt.Errorf("invalid internal port: %w", err)
-			}
+			if strings.ContainsRune(port, '/') {
+				split := strings.SplitN(port, "/", 2)
+				if len(split) != 2 {
+					return nil, fmt.Errorf("malformed port expeted format PORT:DEST/HANDLER[,HANDLER]")
+				}
+
+				for _, handler := range strings.Split(split[1], "+") {
+					h := kraftcloudservices.Handler(handler)
+					if !slices.Contains(kraftcloudservices.Handlers(), h) {
+						return nil, fmt.Errorf("unknown handler: %s (choice of %v)", handler, kraftcloudservices.Handlers())
+					}
+
+					service.Handlers = append(service.Handlers, h)
+				}
 
-			externalPort, err = strconv.Atoi(ports[1])
-			if err != nil {
-				return fmt.Errorf("invalid external port: %w", err)
+				port = split[0]
 			}
-		} else {
-			port, err := strconv.Atoi(opts.Port)
-			if err != nil {
-				return fmt.Errorf("could not parse port number: %w", err)
+
+			if strings.ContainsRune(port, ':') {
+				ports := strings.Split(port, ":")
```

Comment:
SplitN(port, ":", 2)

Reply:
In fact, `SplitN` will guarantee 2 always if there is at least 1 `:`. The existing `SplitN` instances should actually be removed.

Reply:
If you also want to do safety checks, then yes, Split is fine